File: benchmark/output/python-pygame/openai/inventory_grid.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Inventory Management")
clock = pygame.time.Clock()

inv_items = [0.0 for _ in range(20)]
held_item = 0.0

def draw_rect(x, y, w, h):
    pygame.draw.rect(screen, (255, 255, 255), (x, y, w, h), 2)

def draw_circle(x, y, radius):
    pygame.draw.circle(screen, (255, 255, 255), (int(x), int(y)), radius, 2)

def draw_text(text, x, y):
    font = pygame.font.Font(None, 36)
    text_surface = font.render(text, True, (255, 255, 255))
    screen.blit(text_surface, (x, y))

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.KEYDOWN:
            if event.key == pygame.K_1 or event.key == pygame.K_2 or event.key == pygame.K_3:
                item_type = 0.0
                if event.key == pygame.K_1:
                    item_type = 1.0
                elif event.key == pygame.K_2:
                    item_type = 2.0
                elif event.key == pygame.K_3:
                    item_type = 3.0
                for i in range(20):
                    if inv_items[i] == 0.0:
                        inv_items[i] = item_type
                        break
        elif event.type == pygame.MOUSEBUTTONDOWN:
            mx, my = pygame.mouse.get_pos()
            col = int((mx - 200.0) / 80.0)
            row = int((my - 100.0) / 80.0)
            if 0 <= col < 5 and 0 <= row < 4:
                idx = row * 5 + col
                if held_item == 0.0 and inv_items[idx] > 0.0:
                    held_item = inv_items[idx]
                    inv_items[idx] = 0.0
                elif held_item > 0.0 and inv_items[idx] == 0.0:
                    inv_items[idx] = held_item
                    held_item = 0.0
                elif held_item > 0.0 and inv_items[idx] > 0.0:
                    tmp = inv_items[idx]
                    inv_items[idx] = held_item
                    held_item = tmp

    screen.fill((0, 0, 0))
    
    for row in range(4):
        for col in range(5):
            x = 200.0 + col * 80.0
            y = 100.0 + row * 80.0
            draw_rect(x, y, 80.0, 80.0)
            val = inv_items[row * 5 + col]
            if val == 1.0:
                draw_rect(x + 25.0, y + 20.0, 30.0, 8.0)
                draw_rect(x + 37.0, y + 28.0, 6.0, 32.0)
            elif val == 2.0:
                draw_circle(x + 40.0, y + 40.0, 15.0)
            elif val == 3.0:
                draw_circle(x + 40.0, y + 40.0, 8.0)

    draw_text("Held: " + str(int(held_item)), 10.0, 10.0)
    draw_text("1-3: Add | Click: Pick/Place", 200.0, 540.0)

    pygame.display.flip()

pygame.quit()
sys.exit()

File: vibe-lang/benchmark/output/lua-love/gemini/inventory_grid.lua
local inv_items = {}
local held_item = 0.0

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Inventory Management")
    for i = 1, 20 do
        table.insert(inv_items, 0.0)
    end
end

function love.keypressed(key)
    local item_type = 0.0
    if key == "1" then item_type = 1.0
    elseif key == "2" then item_type = 2.0
    elseif key == "3" then item_type = 3.0
    end

    if item_type > 0 then
        for i = 1, 20 do
            if inv_items[i] == 0.0 then
                inv_items[i] = item_type
                break
            end
        end
    end
end

function love.mousepressed(mx, my, button)
    local col = math.floor((mx - 200.0) / 80.0)
    local row = math.floor((my - 100.0) / 80.0)

    if col >= 0 and col < 5 and row >= 0 and row < 4 then
        local idx = row * 5 + col + 1
        if held_item == 0.0 and inv_items[idx] > 0.0 then
            held_item = inv_items[idx]
            inv_items[idx] = 0.0
        elseif held_item > 0.0 and inv_items[idx] == 0.0 then
            inv_items[idx] = held_item
            held_item = 0.0
        elseif held_item > 0.0 and inv_items[idx] > 0.0 then
            local tmp = inv_items[idx]
            inv_items[idx] = held_item
            held_item = tmp
        end
    end
end

function love.draw()
    for row = 0, 3 do
        for col = 0, 4 do
            local x = 200.0 + col * 80.0
            local y = 100.0 + row * 80.0
            
            love.graphics.setColor(1, 1, 1)
            love.graphics.rectangle("line", x, y, 80, 80)
            
            local val = inv_items[row * 5 + col + 1]
            love.graphics.setColor(0.7, 0.7, 0.7)
            if val == 1.0 then
                love.graphics.rectangle("fill", x + 25, y + 20, 30, 8)
                love.graphics.rectangle("fill", x + 37, y + 28, 6, 32)
            elseif val == 2.0 then
                love.graphics.circle("fill", x + 40, y + 40, 15)
            elseif val == 3.0 then
                love.graphics.circle("fill", x + 40, y + 40, 8)
            end
        end
    end
    
    love.graphics.setColor(1, 1, 1)
    love.graphics.print("Held: " .. math.floor(held_item), 10, 10)
    love.graphics.print("1-3: Add | Click: Pick/Place", 200, 540)
end